controllers/util/storagecluster.go

```diff
@@ -0,0 +1,68 @@
+package util
+
+import (
+	"context"
+	"fmt"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+
+	nbv1 "github.com/noobaa/noobaa-operator/v5/pkg/apis/noobaa/v1alpha1"
+	rookCephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type AvailableServices struct {
+	Rbd    bool
+	CephFs bool
+	Nfs    bool
+	Mcg    bool
+}
+
+func GetAvailableServices(
+	ctx context.Context,
+	cl client.Client,
+	storageCluster *ocsv1.StorageCluster,
+) (*AvailableServices, error) {
+	availableServices := &AvailableServices{}
+
+	// rbd - list blockpools and filter out the builtin ones, isNfsAvailable -
+	blockPoolList := &rookCephv1.CephBlockPoolList{}
+	if err := cl.List(ctx, blockPoolList, client.InNamespace(storageCluster.Namespace)); err != nil {
+		return nil, err
+	}
+	for i := range blockPoolList.Items {
+		poolName := blockPoolList.Items[i].Name
+		if poolName != "builtin-mgr" && poolName != GenerateNameForCephNFSBlockPool(storageCluster) {
```

Comment:
Doesn't it make more sense to just check if the RBD default block pool exists?

Reply:
No, as the default blockpool might be disabled and a custom one might be created

Reply:
This is an edge case. But regardless I don't think there is a way cheph without RBD as it is needed for other services to function. in that case let's remove the conditions for rbd here